packages/shared/lib/services/plans/plans.ts

```diff
@@ -0,0 +1,50 @@
+import ms from 'ms';
+
+import type { DBPlan } from '@nangohq/types';
+import type { Knex } from 'knex';
+
+export const TRIAL_DURATION = ms('15days');
+
+export async function getPlan(db: Knex, { accountId }: { accountId: number }): Promise<DBPlan | null> {
+    const res = await db.from<DBPlan>('plans').select<DBPlan>('*').where('account_id', accountId).first();
+    return res || null;
+}
+
+export async function createPlan(
+    db: Knex,
+    { account_id, ...rest }: Pick<DBPlan, 'account_id' | 'name'> & Partial<Omit<DBPlan, 'account_id' | 'created_at' | 'updated_at'>>
+): Promise<DBPlan> {
```

Comment:
I would have all public functions to return a `Result` and have a try catch so one doesn't "forget" to handle errors. I know that's not how all the model services have been written but for new ones we should aim for better error handling